core/src/main/java/org/elasticsearch/index/mapper/geo/GeoPointFieldMapperLegacy.java

```diff
@@ -0,0 +1,447 @@
+                                               DoubleFieldMapper lonMapper, StringFieldMapper geohashMapper, MultiFields multiFields, Explicit<Boolean> ignoreMalformed,
+                                               CopyTo copyTo) {
+            fieldType.setTokenized(false);
+            setupFieldType(context);
+            fieldType.setHasDocValues(false);
+            defaultFieldType.setHasDocValues(false);
+            return new GeoPointFieldMapperLegacy(simpleName, fieldType, defaultFieldType, indexSettings, pathType, latMapper, lonMapper,
+                    geohashMapper, multiFields, ignoreMalformed, coerce(context), copyTo);
+        }
+
+        @Override
+        public GeoPointFieldMapperLegacy build(BuilderContext context) {
+            return super.build(context);
+        }
+    }
+
+    public static Builder parse(Builder builder, Map<String, Object> node, Mapper.TypeParser.ParserContext parserContext) throws MapperParsingException {
+        final boolean indexCreatedBeforeV2_0 = parserContext.indexVersionCreated().before(Version.V_2_0_0);
+        for (Iterator<Map.Entry<String, Object>> iterator = node.entrySet().iterator(); iterator.hasNext();) {
+            Map.Entry<String, Object> entry = iterator.next();
+            String propName = Strings.toUnderscoreCase(entry.getKey());
+            Object propNode = entry.getValue();
+            if (indexCreatedBeforeV2_0 && propName.equals("validate")) {
+                builder.ignoreMalformed = !XContentMapValues.nodeBooleanValue(propNode);
+                iterator.remove();
+            } else if (indexCreatedBeforeV2_0 && propName.equals("validate_lon")) {
+                builder.ignoreMalformed = !XContentMapValues.nodeBooleanValue(propNode);
+                iterator.remove();
+            } else if (indexCreatedBeforeV2_0 && propName.equals("validate_lat")) {
+                builder.ignoreMalformed = !XContentMapValues.nodeBooleanValue(propNode);
+                iterator.remove();
+            } else if (propName.equals(Names.COERCE)) {
+                builder.coerce = XContentMapValues.nodeBooleanValue(propNode);
+                iterator.remove();
+            } else if (indexCreatedBeforeV2_0 && propName.equals("normalize")) {
+                builder.coerce = XContentMapValues.nodeBooleanValue(propNode);
+                iterator.remove();
+            } else if (indexCreatedBeforeV2_0 && propName.equals("normalize_lat")) {
+                builder.coerce = XContentMapValues.nodeBooleanValue(propNode);
+                iterator.remove();
+            } else if (indexCreatedBeforeV2_0 && propName.equals("normalize_lon")) {
+                builder.coerce = XContentMapValues.nodeBooleanValue(propNode);
+                iterator.remove();
+            }
+        }
+        return builder;
+    }
+
+    public static final class GeoPointFieldType extends BaseGeoPointFieldType {
```

Comment:
Good catch! It can be renamed and used in both derived classes. This code is a by-product of once removing the ```lat_lon``` option. Technically ```lat_lon``` is no longer needed, but I kept it around in this initial release for 1. this change is big enough already, and 2. discuss whether we want to leave it as an option to index full precision lat/lon and keep it accessible via ```.lat``` and ```.lon``` extension. In all honesty I think we can remove it altogether in GeoPointV2, it doesn't really buy anything for the amount of space it needs.